### PR TITLE
Remove deprecated `updateSemantics` APIs.

### PR DESCRIPTION
**Pull request by a-wallen**
## fixes https://github.com/flutter/flutter/issues/112221

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/ui/platform_dispatcher.dart [hunk 1]
```diff
@@ -750,27 +750,6 @@ class PlatformDispatcher {
...
 
-  /// Change the retained semantics data about this platform dispatcher.
-  ///
-  /// If [semanticsEnabled] is true, the user has requested that this function
-  /// be called whenever the semantic content of this platform dispatcher
-  /// changes.
-  ///
-  /// In either case, this function disposes the given update, which means the
-  /// semantics update cannot be used further.
-  @Deprecated('''
```
[the remark this answers is not in the document]
Comment (a-wallen) [reply lib/ui/platform_dispatcher.dart:761]: No this has not been deprecated for over a year. That's a good catch. Thanks, Greg!

Is the solution to let the PR sit until this year passes?

**Comment by gspencergoog** [comment]
No, but I'm surprised that the deprecation notice doesn't include the version that the deprecation first appeared in.  We have a lint for that on the Framework side. The deprecation notice should look [like this](https://github.com/flutter/flutter/wiki/Tree-hygiene#deprecations).

Once it has the version number in it, then we'll be able to programmatically detect that it is a year old, and we'll remove it at that time (this is the kind of thing that @Piinks has been doing recently).

Kate, are we supposed to also have deprecation lint checks (for deprecation message format compliance) in the engine `dart:ui` code?

**Review by Piinks — COMMENTED** [review]
How long has this been deprecated for? 
The analyzer-enforced deprecation format only applies in flutter/flutter right now I think. Same for the deprecation policy, we have not applied it to flutter/engine.

We have been discussing expanding to other packages like `dart:ui` , but since we haven't yet we probably shouldn't remove this right now. Especially if it has not been announced or had a migration guide generated yet.

**Comment by a-wallen** [comment]
@Piinks, it's been deprecated since https://github.com/flutter/engine/pull/36675, about one month.

**Comment by gspencergoog** [comment]
@Piinks I think we should probably extend the deprecation notice checks to the `dart:ui` package, since as far as our customers are concerned, it's part of Flutter's Dart code (they probably consider it to be covered by our deprecation policy). What do you think?

**Comment by Piinks** [comment]
> @Piinks I think we should probably extend the deprecation notice checks to the dart:ui package

Totally, that is something we have been discussing. It is just part of the custom analyzer script that runs on CI.

**Comment by Piinks** [comment]
> @Piinks, it's been deprecated since #36675, about one month.

So the warning has not even shipped to stable yet. We should probably wait so folks have time to migrate.

**Comment by skia-gold** [comment]
Gold has detected about 2 new digest(s) on patchset 2.
View them at https://flutter-engine-gold.skia.org/cl/github/37396

**Pull request closed without merging**